docs/developer-docs/integrations/sns/managing/making-proposals.md

```diff
@@ -463,8 +463,52 @@ Typically a generic proposal will have the following structure: a developer send
 * [`RemoveGenericNervousSystemFunction`](#removegenericnervoussystemfunction).
 * [`ExecuteGenericNervousSystemFunction`](#executegenericnervoussystemfunction).
 
+### Defining a generic proposal
+
+A generic proposal is defined by two parts:
+
+1. **A target method and canister** (respectively called `target_method_name` and `target_canister_id` in the code): This is the method that will be called if this generic proposal is adopted. A community can implement any behavior in a proposal by writing within a target method on a canister, then registering that target method in a generic proposal.
+
+2. **A validator method and canister** (respectively called `validator_method_name` and `validator_canister_id` in the code): Since the governance canister is not aware of what a generic proposal does or in which context it will be applied, it cannot validate the proposal’s payload. Therefore, to check whether a proposal’s payload is valid at proposal submission time, the SNS community must implement this validation in a separate method (this can be on the same canister as the target method or on a different one). This method is then called whenever such a generic proposal is submitted.
```

Comment:
```suggestion
2. **A validator method and canister** (respectively called `validator_method_name` and `validator_canister_id` in the code): Since the governance canister is not aware of what a generic proposal does or in which context it will be applied, it cannot validate the proposal’s payload. Therefore, to check whether a proposal’s payload is valid at proposal submission time, the SNS community must implement this validation in a separate method (this can be on the same canister as the target method or on a different one). This method is then called whenever such a generic proposal is submitted. If the validator method fails, the proposal will not put to vote in the SNS.
```